.github/workflows/push_container.yaml

```diff
@@ -28,17 +32,35 @@ jobs:
     needs: [unit-test]
 
     steps:
-    - name: checkout
-      uses: actions/checkout@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
-    - name: get version tag
-      run: echo TAG_NAME=$(echo ${GITHUB_REF} | rev | cut -d"/" -f1 | rev) >> $GITHUB_ENV
+      - name: Log in to the Container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-    - name: docker login
-      run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-    - name: docker build
-      run: >-
-        docker buildx create --name builder &&
-        docker buildx use --builder builder &&
-        make dockerx-build version=${TAG_NAME}
+      - name: Build and push Docker image
```

Comment:
Can we add a copy of this step to the PR workflow with `push` set to false? Would be nice to know if the build fails on any of the architectures before we try to release a new version.